lib/whitehall/publishing_api/populator.rb

```diff
@@ -0,0 +1,66 @@
+module Whitehall
+  class PublishingApi
+    class Populator
+      attr_reader :items, :logger, :sender
+
+      def initialize(items:, sender:, logger: Logger.new(nil))
+        @logger = logger
+        @progress_logger = ProgressLogger.new(logger)
+        @items = items
+        @sender = sender
+      end
+
+      def call
+        items.each do |item|
+          @progress_logger.log(item)
+          sender.call(item)
+        end
+
+        logger.info "Finished."
+      end
+
+      def self.default_items
+        Enumerator.new do |yielder|
+          edition_scope.find_each do |edition|
```

Comment:
Should provide a default for edition_scope on the base class, even if it just raises NotImplementedError.